File: Client/app/src/main/java/ua/nure/havrysh/kindergarten/activities/HumanActivity.java
package ua.nure.havrysh.kindergarten.activities;

import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;

import kindergarden.hakito.kindergardenclient.R;
import kindergarden.hakito.kindergardenclient.databinding.ActivityHumanBinding;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import ua.nure.havrysh.kindergarten.model.Human;
import ua.nure.havrysh.kindergarten.rest.Rest;

public class HumanActivity extends AppCompatActivity {

    public static final String EXTRA_HUMAN_ID = "parentId";

    Human h;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        final ActivityHumanBinding binding = DataBindingUtil.setContentView(this, R.layout.activity_human);
        long humanId = getIntent().getLongExtra(EXTRA_HUMAN_ID, -1);

        if (humanId != -1) {
            Rest.get().getHuman(humanId).enqueue(new Callback<Human>() {
                @Override
                public void onResponse(Call<Human> call, Response<Human> response) {
                    h = response.body();
                    binding.setHuman(h);
                }

                @Override
                public void onFailure(Call<Human> call, Throwable t) {

                }
            });
        } else {
            h = new Human();
            binding.setHuman(h);
        }


    }
}

File: Client/app/src/main/java/ua/nure/havrysh/kindergarten/activities/HumanActivity.kt
package ua.nure.havrysh.kindergarten.activities

import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import kindergarten.hakito.kindergartenclient.R
import kindergarten.hakito.kindergartenclient.databinding.ActivityHumanBinding
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import ua.nure.havrysh.kindergarten.App
import ua.nure.havrysh.kindergarten.model.Human
import ua.nure.havrysh.kindergarten.rest.Rest

class HumanActivity : AppCompatActivity() {
    
    lateinit var human: Human
    lateinit var binding: ActivityHumanBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_human)
        val humanId = intent.getStringExtra(EXTRA_HUMAN_ID)
        
        if (humanId.isNotEmpty()) {
            loadHuman(humanId)
        } else {
            human = Human()
            binding.human = human
        }
    }
    
    private fun loadHuman(id: String) {
        App.scope.launch(Dispatchers.IO) {
            human = Rest.get().getHuman(id).await()
            withContext(Dispatchers.Main) {
                binding.human = human
            }
        }
    }
    
    companion object {
        
        val EXTRA_HUMAN_ID = "parentId"
    }
}
